router/underlayproviders/udpip/udpip.go

```diff
@@ -382,28 +445,38 @@ type externalLink struct {
 // don't keep the remote address, but in the future, we will be making the connections (and
 // the conn argument will be gone), so we will need the address for that.
 func (u *provider) NewExternalLink(
-	conn router.BatchConn,
 	qSize int,
 	bfd *bfd.Session,
-	remote netip.AddrPort,
+	local string,
+	remote string,
 	ifID uint16,
 	metrics router.InterfaceMetrics,
 ) (router.Link, error) {
 
-	if remote == (netip.AddrPort{}) {
-		// The router doesn't do this. This is an internal error.
-		panic("Zero address not supported")
+	localAddr, err := conn.ResolveAddrPortOrPort(local)
+	if err != nil {
+		return nil, err
+	}
+	remoteAddr, err := conn.ResolveAddrPort(remote)
+	if err != nil {
+		return nil, err
 	}
 
```

Comment:
```suggestion
	localAddr, err := conn.ResolveAddrPortOrPort(local)
	if err != nil {
		return nil, fmt.Errorf("resolving local addr: %w")
	}
	remoteAddr, err := conn.ResolveAddrPort(remote)
	if err != nil {
		return nil, fmt.Errorf("resolving remote addr: %w")
	}
```
I suppose this code isn't performance-critical and the potential additional wrapping cost is irrelevant but would improve debuggability.

Reply:
Done.
<!-- Reviewable comment -OMBzlMJ99lUVF7GMKKA:b-896fix -->
<!-- Sent from Reviewable.io -->

Reply:
Done.
<!-- Reviewable comment -OMBzlMJ99lUVF7GMKKA:b-896fix -->
<!-- Sent from Reviewable.io -->